src/library/onlineStatusContext.js

```diff
@@ -50,9 +51,17 @@ const OnlineStatusProvider = ({ children, value }) => {
     return () => {
       stopPingingApi()
     }
-  }, [isNavigatorOnline, stopPingingApi])
+  }, [isNavigatorOnline])
 
   const toggleUserOnlineStatusOverride = () => {
+    const isTheUserAboutToTurnTheAppOnline = hasUserTurnedAppOffline // this is still old state, we havent updated anything yet, thats why its the opposite of what you would expect.
```

Comment:
this is counterintuitive because the `hasUserTurnedAppOffline` is about to be/is stale state so it reads funny

Reply:
how do we know if the user is about to turn the app online? are we referring to the app connection state here? if so can we make the var more specific to that? `isPoorHealthCheckDetected`